driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java

```diff
@@ -159,7 +159,7 @@ static final class WriteOperation extends Operation {
 
     private final Selector selector;
 
-    final ExecutorService executor;
+    private final ExecutorService executor;
```

Comment:
@stIncMale I can roll back the finals. The reduced visibility refactor would ensure that (in the near future) the user-set executor is not shut down from outside this class. Up to you - is this worth the potential future merge conflict?

Reply:
This looks like something worth discussing the team on the catchup meeting, or asking Jeff/Ross here, as they are likely the only ones who have idea on our process of syncing the `tlschannel` code.

Reply:
Force rebased. Removed finals commit, kept visibility commit.